libs/langchain/langchain/callbacks/streaming_aiter.py

```diff
@@ -35,7 +35,11 @@ async def on_llm_new_token(self, token: str, **kwargs: Any) -> None:
             self.queue.put_nowait(token)
 
     async def on_llm_end(self, response: LLMResult, **kwargs: Any) -> None:
-        self.done.set()
+        if (
```

Comment:
some comment about why this is important would be fairly helpful

Also, just so i understand, this happens when a function is called? and therefor text is not provided?

Reply:
Yes. The issue occurs when using function calling, I think that no text is generated during the execution of the function. This can prematurely end the `aiter` process.